hack/build-flags.sh

```diff
@@ -27,13 +28,18 @@ function build_flags() {
     [[ -n "${commit}" ]] || abort "error getting the current commit"
     version="v$(date +%Y%m%d)-local-${commit}"
   fi
-  # Extract Eventing and Serving versions from go.mod
+  # Eventing and Serving versions are set the same as the client version.
+  # For patch version release, they are from go.mod
   local version_serving version_eventing
-  version_serving=$(grep "knative.dev/serving " "${base}/go.mod" \
-    | sed -s 's/.* \(v.[\.0-9]*\).*/\1/')
-  version_eventing=$(grep "knative.dev/eventing " "${base}/go.mod" \
-    | sed -s 's/.* \(v.[\.0-9]*\).*/\1/')
-
+  if [[ "${patch}" -ne 0 ]]; then
```

Comment:
The logic below would imply that for a version `1.2.0` you get a serving and eventing version `0.29.0`, while for `1.2.1` you get `1.2.1` for eventing and serving (regardless of what the actual versions of eventing and serving are). This is not as it should work.

Instead, the algorithm should work like this:

* `$version` is `${TAG}`
* For the serving version:
  - Take `major.minor` from `${version}` (via e.g. `cut -f1-2 -d.`)
  - Take `patch` from the serving dependency in go.mod (e.g. `echo $version_serving | cut -f3 -d.` with `$version_serving` being the full version extracted from go.mod ad it is now) 
  - Combine all of them to calculate the serving version (`major.minor` from `${version}` + `.patch` from the go.mod dependency.

Same for  eventing.

Reply:
Thanks @rhuss @dsimansk ,
I misunderstood the logic. I have updated my implementation in the shell script. Should I do it in version.go?

Reply:
@kobayashi sorry for the delay. I have tested the changes locally and it produces desired results imo. Thanks for the fix!